### PR TITLE
Feature/add events data

### PR DESCRIPTION
**Pull request by BarryGeeMOJ**
Related to the following tickets:

- https://dsdmoj.atlassian.net/browse/LNP-44
- https://dsdmoj.atlassian.net/browse/LNP-75

Changes made in this PR:

- Added prison API call for authenticated user events
- Added events data parsing
- Added date formatting
- Added supporting unit tests
- Code refactoring

### CODE REVIEW

File: server/@types/launchpad/index.d.ts [hunk 1]
```diff
@@ -1,3 +1,25 @@
+export type RawPrisonerEvent = {
```
Comment (psoleckimoj) [inline server/@types/launchpad/index.d.ts:1]: You can probably import the types from the Swagger. You can use `npx` like BAPV shows https://github.com/ministryofjustice/book-a-prison-visit-staff-ui#imported-types and https://prison-api-dev.prison.service.justice.gov.uk/swagger-ui/index.html#/ and it'll auto-generate a file for you to include.

Then you can make easier to use types like this https://github.com/ministryofjustice/hmpps-developer-portal/blob/main/server/data/strapiApiTypes.ts

Saves you trying to keep track of events from other APIs.

**Review by psoleckimoj — DISMISSED** [review]
LGTM!

File: server/@types/launchpad/index.d.ts (hunk 1 above)
Comment (BarryGeeMOJ) [reply server/@types/launchpad/index.d.ts]: Have updated the code to reflect this.

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Pull request merged**